### DISCUSSION TITLE
Consider switching `refocus` logic to utilize Page Visibility API

### DISCUSSION THREAD

**Discussion by stramel** [urql-graphql/urql#1055] [Ideas]
Hello 👋, I was looking into the `urql` implementation for the `refocus` exchange (`@urql/exchange-refocus`). I noticed that it utilizes `window.addEventListener('focus', ...)` for handling the focus. I found a few issues with this implementation:

- `iframe` can cause the `focus` event to fire numerous times if the user interacts with the `iframe` and then back with the current document
- DevTools can cause the `focus` event to fire numerous times if the user/developer interacts with the DevTools and then back with the document. (A very common case for developers)
- A developer fires a `focus` event (ie. `node.dispatchEvent(new CustomEvent('focus', { bubbles: true}))`). This case seems less likely but still a side-effect of the current implementation.

I'm curious if you have considered utilizing the [Page Visibility API](https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API) to achieve a more true implementation for "refocus" of the window. 

Thank you for taking the time to consider my suggestion!

**Comment by kitten** [comment]
Hiya :wave: Sorry for the late reply! I've created an RFC for this change and will see whether someone has time soon to look into this! It's a great suggestion! https://github.com/FormidableLabs/urql/issues/1063

**Reply by stramel** [reply]
Thank you for creating an RFC for this! I look forward to seeing how the discussion progresses.

**Reply by stramel** [reply]
@kitten Is there anything I can do to help contribute to the RFC?

**Reply by kitten** [reply]
Hm, I mean there are some open questions around IE 11 fallbacks / support so I’d love some comments if you’ve dealt with that before or have some ideas on how we could approach that ✌️

Anything that adds to the RFC will make it easier for whomever picks it up to get started! 🙇‍♂️
